HW 3

KobrinN opened the pull request:
-Добавил новый функционал в класс Person из ДЗ 1-2.
-Сделал ДЗ 3.

KobrinN committed:
Исправление ДЗ 1-2
- добавил в классы ДЗ 1-2 методы equals() и hashCode();
- добавил java doc к некоторым методам этих классов:

KobrinN committed:
ДЗ 3
-немного изменил java doc классах ДЗ 1-2.
-добавил два конструктора в класс Person из ДЗ 1-2
-добавил в класс Person метод Person.printPerson() для вывода всех инфомации о данном человеке.
-определил класс Registry c необходимыми полями и методами.
-определил класс Solution в папке lesson3 для проверки работы класса Registry, при этом в реестре хранились записи типа Person из ДЗ 1-2.

lefasdayan commented on ru/turing/courses/lesson2/animals/Person.java:
```diff
@@ -16,77 +15,157 @@ public class Person {
+     * @param regexBirthday формат строки, содержащей дату рождения.
+     * @param address адрес проживания человека.
+     * @param regexAddress формат адреса проживания человека.
+     */
```
<3 за javadoc

lefasdayan commented on ru/turing/courses/lesson3/Registry.java:
```diff
@@ -1,13 +1,85 @@
+ * @param <Key>   тип ключа реестра.
+ * @param <Value> тип значения записи реестра.
+ */
+public class Registry<Key extends Number, Value extends Person> {
```
```suggestion
public class Registry<Integer, Value extends Person> {
```
Прописываешь для ключа Key extends Number, а потом в мэйне запрашиваешь int. Не легче сразу прописать Integer?)

lefasdayan commented on ru/turing/courses/lesson3/Registry.java:
```diff
@@ -1,13 +1,85 @@
+    public Value removeByKey(Key key) {
+        return this.storage.remove(key);
+    }
+
```
removeByValue() еще

KobrinN committed:
Исправил ДЗ 3
-исправил все java doc(надеюсь что все).
-добавил Registry.removeByKey.
-добавил проверку в Solution нового метода.

lefasdayan commented:
Отлично!
